src/generate.ts

```diff
@@ -146,10 +156,12 @@ export class Generate {
     if (devfileMetadata.generateName) {
       devWorkspaceMetadata.generateName = devfileMetadata.generateName;
     }
-    if (addDevfileContent) {
-      devWorkspaceMetadata.annotations = {
-        'che.eclipse.org/devfile': jsYaml.dump(devfile),
-      };
+    if (devfile.attributes?.[DEVWORKSPACE_METADATA_ANNOTATION]) {
+      devWorkspaceMetadata.annotations = Object.assign({}, devfile.attributes?.[DEVWORKSPACE_METADATA_ANNOTATION]);
+      delete devfile.attributes?.[DEVWORKSPACE_METADATA_ANNOTATION];
```

Comment:
you can skip `?.` here, delete doesn't throw an error if the property doesn't exist

Reply:
@svor If the property doesn't exist, the devworkspace is not managed by CHE.

Reply:
I have fixed it:
```js
    if (devfile.attributes?.[DEVWORKSPACE_METADATA_ANNOTATION]) {
      devWorkspaceMetadata.annotations = Object.assign({}, devfile.attributes[DEVWORKSPACE_METADATA_ANNOTATION]);
      delete devfile.attributes[DEVWORKSPACE_METADATA_ANNOTATION];
      if (Object.keys(devfile.attributes).length === 0) {
        delete devfile.attributes;
      }
    }
```